modules/calib3d/src/ptsetreg.cpp

```diff
@@ -505,6 +505,93 @@ class Affine3DEstimatorCallback : public PointSetRegistrator::Callback
     }
 };
 
+
+/*
+ * Compute
+ *  x      X     t1
+ *  y  =   Y  +  t2
+ *  z      Z     t3
+ *
+ *  - every element in _m1 contains (X,Y,Z), which are called source points
+ *  - every element in _m2 contains (x,y,z), which are called destination points
+ *  - _model is of size 3x1, which contains
+ *      t1
+ *      t2
+ *      t3
+ */
+class Translation3DEstimatorCallback : public PointSetRegistrator::Callback
+{
+public:
+    int runKernel( InputArray _m1, InputArray _m2, OutputArray _model ) const CV_OVERRIDE
+    {
+
+        Mat m1 = _m1.getMat(), m2 = _m2.getMat();
+        const Point3f* from = m1.ptr<Point3f>();
+        const Point3f* to   = m2.ptr<Point3f>();
+
+        double buf[3];
+        Mat T(3, 1, CV_64F, &buf[0]);
+        double* Tptr = T.ptr<double>();
```

Comment:
Consider using `Matx13d` instead.

```.diff
-T.copyTo(_model);
+Mat(T, false).copyTo(_model);
```

Reply:
Done! Also required modifying `test_translation3d_estimator` to use the new data type